couchbase/operations/document_analytics.cxx

```diff
@@ -173,38 +163,36 @@ analytics_request::make_response(error_context::analytics&& ctx, const encoded_r
             bool link_not_found = false;
             bool compilation_failure = false;
 
-            if (response.meta.errors) {
-                for (const auto& error : *response.meta.errors) {
-                    switch (error.code) {
-                        case 21002: /* Request timed out and will be cancelled */
-                            server_timeout = true;
-                            break;
-                        case 23007: /* Job queue is full with [string] jobs */
-                            job_queue_is_full = true;
-                            break;
-                        case 24044: /* Cannot find dataset [string] because there is no dataverse declared, nor an alias with name [string]!
-                                     */
-                        case 24045: /* Cannot find dataset [string] in dataverse [string] nor an alias with name [string]! */
-                        case 24025: /* Cannot find dataset with name [string] in dataverse [string] */
-                            dataset_not_found = true;
-                            break;
-                        case 24034: /* Cannot find dataverse with name [string] */
-                            dataverse_not_found = true;
-                            break;
-                        case 24040: /* A dataset with name [string] already exists in dataverse [string] */
-                            dataset_exists = true;
-                            break;
-                        case 24039: /* A dataverse with this name [string] already exists. */
-                            dataverse_exists = true;
-                            break;
-                        case 24006: /* Link [string] does not exist | Link [string] does not exist */
-                            link_not_found = true;
-                            break;
-                        default:
-                            if (error.code >= 24000 && error.code < 25000) {
-                                compilation_failure = true;
-                            }
-                    }
+            for (const auto& error : response.meta.errors) {
```

Comment:
In the previous PR, you modify error handling to use the first_error_code and what not for error handling, but in this PR you loop over them.  I don't think a loop makes sense here since we can only return a single error code.

Reply:
it is not previous PR, it is subsequent, which will come after this one